app/services/container_topology_service.rb

```diff
@@ -117,35 +117,30 @@ def build_entity_data(entity)
     data
   end
 
-  def entity_status(entity, kind)
-    case kind
-    when 'Vm', 'Host' then entity.power_state.capitalize
-    when 'ContainerNode'
-      ready_status = 'Unknown'
+  def entity_status(entity)
+    if entity.kind_of?(Host) || entity.kind_of?(Vm)
+      status = entity.power_state.capitalize
+    elsif entity.kind_of?(ContainerNode)
+      status = 'Unknown'
       entity.container_conditions.each do |condition|
         if condition.try(:name) == 'Ready' && condition.try(:status) == 'True'
-          ready_status = 'Ready'
+          status = condition.name
         else
-          ready_status = 'NotReady'
+          status = 'NotReady'
```

Comment:
OK the logic here is as following: There isn't first or last one here, there *can* be a condition named 'Ready', and it can get true or false values. So if there is a condition named 'Ready' and its value is 'true' - then the status is ready. If there is a condition named Ready and its status is not true - it's interpreted that the node is not ready. If the conditions don't contain a condition named Ready, then we can't conclude anything about the status hence it is unknown. I was never really fond of the way this status was reported as a condition, but this is the k8s side design.